src/modules/mavlink/mavlink_mission.cpp

```diff
@@ -305,19 +305,19 @@ void
 MavlinkMissionManager::send_mission_item(uint8_t sysid, uint8_t compid, uint16_t seq)
 {
 	mission_item_s mission_item = {};
-	bool read_success = false;
+	int read_result = 0;
 
 	switch (_mission_type) {
 
 	case MAV_MISSION_TYPE_MISSION: {
-			read_success = dm_read(_dataman_id, seq, &mission_item, sizeof(mission_item_s)) == sizeof(mission_item_s);
+			read_result = dm_read(_dataman_id, seq, &mission_item, sizeof(mission_item_s)) == sizeof(mission_item_s);
```

Comment:
Now I'm confused. Before this commit it was `bool = size_t == size_t`
With this commit it became `int = size_t == size_t` which is according to my understanding an implicit cast from bool (result of == operator) to int.
So if I'm not mistaken this commit does the opposite of what it intended to do.
We should probably set the parentheses differently for the operations to be visible more clearly.

Reply:
Here's my suggestion to make this more clear: https://github.com/PX4/PX4-Autopilot/pull/19653

Reply:
Oops.